cli/src/main/java/org/jboss/as/cli/handlers/ShutdownHandler.java

```diff
@@ -172,28 +171,40 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
             disconnect = false;
         }
 
+        // Check if we are using a client launched from the server/host installation we want shutdown to perform an update.
+        // In such a case, we will exit from the current JBoss CLI process to avoid interfering with the server/host update.
+        // This will also force the user to relaunch the CLI session using the most recent updates once the server/host has
+        // been updated.
+        // The shutdown mgmt Operation will return the content of the same client marker file created by this CLI instance if
+        // both are using the same server installation. Otherwise, no file marker value will be returned by the ShutDown operations.
+        // This check is only relevant when we are performing a installation.
+        ModelNode clientMarker = executeOperation(client, cliClient, op, true);
         if (Util.TRUE.equalsIgnoreCase(performInstallation.getValue(ctx.getParsedCommandLine()))) {
-            // Check if I am using a client launched from the server/host installation we want to shutdown to perform an update.
-            // In such a case, we will exit from the current JBoss CLI process to avoid interfering with the server/host update.
-            // This will also force the user to relaunch the CLI session using the most recent updates once the server/host has
-            // been updated.
-            boolean localClientLaunch = isLocalClientLaunch(ctx);
-            executeOperation(client, cliClient, op, !localClientLaunch);
-            if (localClientLaunch) {
+            boolean isLocalClient = true;
+            if (clientMarker != null) {
+                final String jbossHome = WildFlySecurityManager.getEnvPropertyPrivileged("JBOSS_HOME", null);
+                final String clientMarkerData = WildFlySecurityManager.getPropertyPrivileged(Util.CLI_MARKER_VALUE, null);
+                if (jbossHome != null && clientMarkerData != null) {
```

Comment:
Not sure why we need to check for the presence of JBOSS_HOME env variable?

Reply:
Right, it was left over from a previous approach, we can remove the JBOSS_HOME check here